### PR TITLE
Extract BLS library used in polybft to a standalone package

### PR DESCRIPTION
**Pull request by Stefan-Ethernal**
# Description

This PR extracts bls signing abstractions already used for `polybft` consensus protocol, which relies on this library https://github.com/umbracle/go-eth-bn256 to Edge standalone package.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist

- [ ] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### Manual tests

Please complete this section if you ran manual tests for this functionality, otherwise delete it

# Documentation update

Please link the documentation update PR in this section if it's present, otherwise delete it

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it

### CODE REVIEW

File: consensus/polybft/signer/utils_test.go [hunk 1]
```diff
@@ -1,99 +0,0 @@
...
-	isOk := signature.Verify(blsKey.PublicKey(), validTestMsg, DomainValidatorSet)
-	assert.True(t, isOk)
-
-	// Verify if invalid message is signed with correct private key. Only use public key for the verification
-	// this should fail => isOk = false
-	isOk = signature.Verify(blsKey.PublicKey(), invalidTestMsg, DomainValidatorSet)
-	assert.False(t, isOk)
-}
-
-func Test_AggregatedSign(t *testing.T) {
```
Comment (igorcrevar) [inline consensus/polybft/signer/utils_test.go:33]: Maybe we should move `Test_AggregatedSign` and `Test_MakeKOSKSignature` to the new file also?

File: helper/staking/staking.go [hunk 2]
```diff
@@ -107,13 +108,15 @@ func setBytesToStorage(
 		return
 	}
 
-	// Set size at the base index
-	baseSlot := types.Hash{}
-	baseSlot[31] = byte(2*dataLen + 1)
-	storageMap[baseIndex] = baseSlot
+	// Store the size of the data in the first slot
+	encodedSize := make([]byte, 32)
+	binary.BigEndian.PutUint64(encodedSize[24:], uint64(2*dataLen+1))
```
Comment (igorcrevar) [inline helper/staking/staking.go]: clever :)

**Review by igorcrevar — APPROVED** [review]
Just a small remark, otherwise good

**Comment by vcastellm** [comment]
If this is not IBFT backwards compatible maybe IBFT can be removed at this point?

**Comment by Stefan-Ethernal** [comment]
> If this is not IBFT backwards compatible maybe IBFT can be removed at this point?

We can discuss it and decide, I was thinking the same. 

However, I'd still prefer to have this PR merged (first), because:
- I find it useful to extract the bls package to a separate one, outside of the polybft, because it should not be tied with any specific consensus protocol,
- removing IBFT entirely represents a larger effort. 

Does it make sense?

File: consensus/polybft/signer/utils_test.go (hunk 1 above)
Comment (Stefan-Ethernal) [reply consensus/polybft/signer/utils_test.go:33]: Fixed https://github.com/0xPolygon/polygon-edge/pull/1981/commits/52365b93c5f0f442f3b84ba3df8bcb47cc8d0292

**Review by dusan-maksimovic — APPROVED** [review]

**Review by stana-miric — APPROVED** [review]

**Review by goran-ethernal — APPROVED** [review]

**Pull request merged**